### ISSUE TITLE
How do I publish a clib that depends on another clib

### ISSUE THREAD

**Issue by thlorenz** [clibs/clib#17]
I wrote this library that depends on clibs/list which is [installs automatcially](https://github.com/thlorenz/ee.c/blob/35ec6223017470f5ab0efa8be9accec585fc5a15/Makefile#L52) during make.

My question is how do I properly publish this and/or what instructions to I provide?

So lets say the user installs my lib via: `clib install ee.c -o deps`
Now in order to use it he'd actually have two options:
- `cd deps/ee.c && make deps/list` which would pull down `list` but then he'd still need to add the `list` related things like [sources](https://github.com/thlorenz/ee.c/blob/35ec6223017470f5ab0efa8be9accec585fc5a15/Makefile#L12) when building his lib
- `cd deps/ee.c && make build/libee.a` if he wants to link it locally

None of these seem like great options to me since then the user has to be aware of how to build the things he depends on, where ideally it should just be as easy as `clib install x/y` and done.

I noticed that `mon` even checks in all deps (instead of installing them).

So basically I'm trying to find a way (possibly via a `postinstall`) to prep the dependency at least to the point where the user does not have to be aware of all the nested dependencies and how to build them in order to use my lib.

Otherwise a lengthy explanation would have to be included with the readme on how to get started in which case just doing a `make install` would be easier although then we are back to installing libs globally (yikes).
